sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/FluxAutoComplete.java

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.logging.ClientLogger;
+import org.reactivestreams.Subscription;
+import reactor.core.CoreSubscriber;
+import reactor.core.publisher.BaseSubscriber;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxOperator;
+import reactor.core.publisher.Mono;
+import reactor.util.context.Context;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * Flux operator that auto-completes or auto-abandons messages when control is returned successfully.
+ */
+class FluxAutoComplete<T extends ServiceBusReceivedMessageContext> extends FluxOperator<T, T> {
```

Comment:
Why do we need a generic type that extends `ServiceBusReceivedMessageContext`? Also, this reminds me that `ServiceBusReceivedMessageContext` should be marked `final` too.